File: flutter_app/fetch_op_tmdb.dart
import 'dart:convert';
import 'dart:io';
import 'package:dio/dio.dart';

void main() async {
  final _dio = Dio();
  final tmdbKey = 'd60e665c617655185db02ee381c7bd0a';
  
  try {
    print('Fetching One Piece (TMDB ID 37854)...');
    final response = await _dio.get(
      'https://api.themoviedb.org/3/tv/37854?api_key=$tmdbKey',
    );
    
    final seasons = response.data['seasons'] as List<dynamic>;
    int totalEpisodes = 0;
    List<Map<String, dynamic>> allEpisodes = [];
    
    for (var season in seasons) {
      final seasonNumber = season['season_number'];
      if (seasonNumber == 0) continue; // Skip specials
      
      print('Fetching season $seasonNumber...');
      final sResponse = await _dio.get(
        'https://api.themoviedb.org/3/tv/37854/season/$seasonNumber?api_key=$tmdbKey',
      );
      
      final eps = sResponse.data['episodes'] as List<dynamic>;
      for (var ep in eps) {
        totalEpisodes++;
        allEpisodes.add({
          'mal_id': totalEpisodes,
          'title': ep['name'],
          'title_japanese': '',
          'aired': ep['air_date'] != null ? ep['air_date'] + "T00:00:00+00:00" : null,
          'filler': false,
          'recap': false,
          'synopsis': ep['overview'],
        });
      }
    }
    
    print('Total absolute episodes fetched: $totalEpisodes');
    File('assets/data/one_piece_episodes.json').writeAsStringSync(json.encode(allEpisodes));
    print('Saved to assets/data/one_piece_episodes.json');
    
  } catch (e) {
    print('Error: $e');
  }
}

File: flutter_app/fetch_op_tmdb.py
import requests
import json
import os

tmdb_key = 'd60e665c617655185db02ee381c7bd0a'
all_episodes = []

try:
    print('Fetching One Piece (TMDB ID 37854)...')
    r = requests.get(f'https://api.themoviedb.org/3/tv/37854?api_key={tmdb_key}')
    r.raise_for_status()
    data = r.json()
    seasons = data['seasons']
    total_episodes = 0
    
    for season in seasons:
        s_num = season['season_number']
        if s_num == 0: continue
        print(f'Fetching season {s_num}...')
        sr = requests.get(f'https://api.themoviedb.org/3/tv/37854/season/{s_num}?api_key={tmdb_key}')
        sr.raise_for_status()
        s_data = sr.json()
        for ep in s_data['episodes']:
            total_episodes += 1
            all_episodes.append({
                'mal_id': total_episodes,
                'title': ep['name'],
                'title_japanese': '',
                'aired': ep['air_date'] + 'T00:00:00+00:00' if ep.get('air_date') else None,
                'filler': False,
                'recap': False,
                'synopsis': ep['overview']
            })
            
    print(f'Total episodes: {total_episodes}')
    with open('assets/data/one_piece_episodes.json', 'w') as f:
        json.dump(all_episodes, f)
    print('Done!')
except Exception as e:
    print(f'Error: {e}')
